src/Security/MemberAuthenticator/LostPasswordHandler.php

```diff
@@ -24,108 +36,120 @@ class MemberLoginHandler extends FormRequestHandler
+            'Email' => $email
+        ];
     }
 
+
     /**
-     * Login in the user and figure out where to redirect the browser.
+     * Factory method for the lost password form
      *
-     * The $data has this format
-     * array(
-     *   'AuthenticationMethod' => 'MemberAuthenticator',
-     *   'Email' => 'sam@silverstripe.com',
-     *   'Password' => '1nitialPassword',
-     *   'BackURL' => 'test/link',
-     *   [Optional: 'Remember' => 1 ]
-     * )
+     * @skipUpgrade
+     * @return Form Returns the lost password form
+     */
+    public function lostPasswordForm()
+    {
+        return LoginForm::create(
+            $this,
+            $this->authenticatorClass,
+            'LostPasswordForm',
+            new FieldList(
+                new EmailField('Email', _t('Member.EMAIL', 'Email'))
+            ),
+            new FieldList(
+                new FormAction(
+                    'forgotPassword',
+                    _t('Security.BUTTONSEND', 'Send me the password reset link')
+                )
+            ),
+            false
+        );
+    }
+
+    /**
+     * Redirect to password recovery form
      *
-     * @param array $data
      * @return HTTPResponse
      */
-    protected function logInUserAndRedirect($data)
+    public function redirectToLostPassword()
     {
-        Session::clear('SessionForms.MemberLoginForm.Email');
-        Session::clear('SessionForms.MemberLoginForm.Remember');
-
-        $member = Member::currentUser();
-        if ($member->isPasswordExpired()) {
-            return $this->redirectToChangePassword();
-        }
-
-        // Absolute redirection URLs may cause spoofing
-        $backURL = $this->getBackURL();
-        if ($backURL) {
-            return $this->redirect($backURL);
-        }
-
-        // If a default login dest has been set, redirect to that.
-        $defaultLoginDest = Security::config()->get('default_login_dest');
-        if ($defaultLoginDest) {
-            return $this->redirect($defaultLoginDest);
-        }
-
-        // Redirect the user to the page where they came from
-        if ($member) {
-            if (!empty($data['Remember'])) {
-                Session::set('SessionForms.MemberLoginForm.Remember', '1');
-                $member->logIn(true);
-            } else {
-                $member->logIn();
-            }
-
-            // Welcome message
-            $message = _t(
-                'Member.WELCOMEBACK',
-                "Welcome Back, {firstname}",
-                ['firstname' => $member->FirstName]
-            );
-            Security::setLoginMessage($message, ValidationResult::TYPE_GOOD);
-        }
+        $lostPasswordLink = Security::singleton()->Link('lostpassword');
+        return $this->redirect($this->addBackURLParam($lostPasswordLink));
+    }
 
-        // Redirect back
-        return $this->redirectBack();
+    public function getReturnReferer()
+    {
+        return $this->link();
```

Comment:
I think that changing it to `getLink()`, like I suggested, would support both approaches. But as @sminnee pointed out, it will take a way bigger refactor and have much more impact. At the moment, I'm inclined to say "go with consistency for now and have a look later"?
What's your thought on that approach @dhensby ?